comments/board.py

```diff
@@ -0,0 +1,441 @@
+# Game_Piece Super & Sub class
+class Game_Piece():
+    def __init__(self, color, number):
+        self.name = color + "_piece_" + str(number)
+        self.xy_coord = None
+        self.team = color
+
+    def update_position(self, xy_coord):
+        self.xy_coord = xy_coord
+
+
+class Checkers_Game_Piece(Game_Piece):
+    def __init__(self, color, number):
+        super().__init__(color, number)
+        self.is_king = False
+        self.moves = {
+            "move_nw": None,
+            "move_ne": None,
+            "move_sw": None,
+            "move_se": None
+        }
+
+    def king_me(self):
+        self.is_king = True
+        return
+    """
+    def demote(self):  # Not currently required
+        self.is_king = False
+        return
+    """
+
+    def check_valid_move(self):
+        self.moves.update({key: None for key in self.moves})  # Reset possible moves before check
+        if (self.xy_coord[0] - 1 >= 1) and (self.xy_coord[1] + 1 <= 8):  # Check NW
+            self.moves["move_nw"] = self.xy_coord[0] - 1, self.xy_coord[1] + 1
+        if (self.xy_coord[0] + 1 <= 8) and (self.xy_coord[1] + 1 <= 8):  # Check NE
+            self.moves["move_ne"] = self.xy_coord[0] + 1, self.xy_coord[1] + 1
+        if (self.xy_coord[0] - 1 >= 1) and (self.xy_coord[1] - 1 >= 1):  # Check SW
+            self.moves["move_sw"] = self.xy_coord[0] - 1, self.xy_coord[1] - 1
+        if (self.xy_coord[0] + 1 <= 8) and (self.xy_coord[1] - 1 <= 8):  # Check SE
+            self.moves["move_se"] = self.xy_coord[0] + 1, self.xy_coord[1] - 1
+        """
+        If coordinate is less than 1 or greater than 8 in x or y 
+        its invalid so remove it and set to None. We hardcode because 
+        board game min/max dimensions are not known to a Checkers_Game_Piece()
+        """
+        for key, xy_coord in self.moves.items():
+            try:
+                if xy_coord is not None and (xy_coord[0] < 1) or (xy_coord[1] < 1) or (xy_coord[0] > 8) or (xy_coord[1] > 8):
+                    self.moves[key] = None
+            except TypeError:
+                """
+                This fixes NoneType object is not subscriptable
+                It occurs if a piece is at the edge of a board
+                """
+                pass
+        # print(f"Debug self.moves {self.moves}")
+        return self.moves
```

Comment:
This is some weird logic. Don't you already do that logic up above here? Cause you're basically checking the same thing in all your if statements above. And if the above if statement doesn't resolve your dictionary value is defaulted to None anyways

Reply:
the if statements can technically have a resultant value of 1,1 or 8,8 before the final assignment to the dict for example at line 40. this would then cause it to become 0,1 / 9,8 etc. at line 41. This catches the edge cases that occur at any coord that exists at 1,1 before assignment to the dict and makes them none values instead.

we could solve this by adding another check during the if statements as well i think.